drivers/libusb.c

```diff
@@ -246,13 +239,8 @@ static int libusb_open(usb_dev_handle **udevp, USBDevice_t *curDevice, USBDevice
 			upsdebugx(2, "- Device: %s", curDevice->Device ? curDevice->Device : "unknown");
 			upsdebugx(2, "- Device release number: %04x", curDevice->bcdDevice);
 
-			/* Powervar UPS is a composite device. Index must be 1 to select the right interface */
-			if ((curDevice->VendorID == 0x4234) && (curDevice->ProductID == 0x0002)) {
-				hid_rep_index = 1;
-				upsdebugx(2, "Set rep index to 1");
-			}
 			if ((curDevice->VendorID == 0x463) && (curDevice->bcdDevice == 0x0202)) {
-				hid_desc_index = 1;
+				usb_subdriver.hid_desc_index = 1;
```

Comment:
@aquette : generally the change in this PR makes sense to me. I wonder if the comment earlier in the original file (where it defined `hid_desc_index` in relation to `bcdDevice`) should be moved here to explain why we assign the index. Currently it has moved to libusb.h...

@abratchik: Even better might be to move this assignment to an eaton-specific subdriver same as this PR did for Powervar UPS (and introduced Arduino support), so the shared `libusb.c` code has less weird cases handled directly.

Reply:
@jimklimov I fully agree but this is a bit more work. I can cleanse this hardcode although it was not in plans :). My main intent was to support Arduino boards but they are composite devices by definition, so they use interface 0 for sketch flashing purposes.

Reply:
@jimklimov I checked the changes required for removing hardcode for Eaton- it is a bit more complex than I thought. Since I don't have Eaton device for testing, I suggest we can leave it as-is, without touching Eaton. Powervar UPS changes were relatively straightforward and harmless so I decided to propose refactoring. Regarding Eaton - one can do it later in a  in a separate PR if required.